Utilities/DataFlow/src/SubframeBuilderDevice.cxx

```diff
@@ -18,63 +18,88 @@
 #include <functional>
 
 #include "DataFlow/SubframeBuilderDevice.h"
+#include "DataFlow/SubframeUtils.h"
 #include "Headers/SubframeMetadata.h"
 #include "Headers/HeartbeatFrame.h"
 #include "Headers/DataHeader.h"
 #include <options/FairMQProgOptions.h>
 
-// From C++11 on, constexpr static data members are implicitly inlined. Redeclaration
-// is still permitted, but deprecated. Some compilers do not implement this standard
-// correctly. It also has to be noticed that this error does not occur for all the
-// other public constexpr members
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mOrbitsPerTimeframe;
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mOrbitDuration;
-constexpr uint32_t o2::DataFlow::SubframeBuilderDevice::mDuration;
-
 using HeartbeatHeader = o2::Header::HeartbeatHeader;
 using HeartbeatTrailer = o2::Header::HeartbeatTrailer;
 using DataHeader = o2::Header::DataHeader;
+using SubframeId = o2::dataflow::SubframeId;
 
 o2::DataFlow::SubframeBuilderDevice::SubframeBuilderDevice()
   : O2Device()
 {
-  LOG(INFO) << "o2::DataFlow::SubframeBuilderDevice::SubframeBuilderDevice " << mDuration << "\n";
 }
 
 o2::DataFlow::SubframeBuilderDevice::~SubframeBuilderDevice()
 = default;
 
 void o2::DataFlow::SubframeBuilderDevice::InitTask()
 {
-//  mDuration = GetConfig()->GetValue<uint32_t>(OptionKeyDuration);
+  mOrbitDuration = GetConfig()->GetValue<uint32_t>(OptionKeyOrbitDuration);
+  mOrbitsPerTimeframe = GetConfig()->GetValue<uint32_t>(OptionKeyOrbitsPerTimeframe);
   mIsSelfTriggered = GetConfig()->GetValue<bool>(OptionKeySelfTriggered);
   mInputChannelName = GetConfig()->GetValue<std::string>(OptionKeyInputChannelName);
   mOutputChannelName = GetConfig()->GetValue<std::string>(OptionKeyOutputChannelName);
+  mFLPId= GetConfig()->GetValue<size_t>(OptionKeyFLPId);
+  mStripHBF= GetConfig()->GetValue<bool>(OptionKeyStripHBF);
 
+  // depending on whether the device is self-triggered or expects input,
+  // the handler function needs to be registered or not.
+  // ConditionalRun is not called anymore from the base class if the
+  // callback is registered
   if (!mIsSelfTriggered) {
```

Comment:
Lets get rid of the 'self triggered' mode and the corresponding option. It doesn't do anything.